e2e/e2e-starter.sh

```diff
@@ -21,24 +31,31 @@ if [ -n "$out" ]; then
     MC_RG_NAME="MC_${RESOURCE_GROUP_NAME}_${CLUSTER_NAME}_$LOCATION"
     exists=$(az group exists -n $MC_RG_NAME)
     if [ $exists = "false" ]; then
-        log "Deleting cluster"
-        clusterDeleteStartTime=$(date +%s)
-        az aks delete -n $CLUSTER_NAME -g $RESOURCE_GROUP_NAME --yes
-        clusterDeleteEndTime=$(date +%s)
-        log "Deleted cluster in $((clusterDeleteEndTime-clusterDeleteStartTime)) seconds"
+        # The cluster exists, but has no associated MC resource group
+        log "Cluster $CLUSTER_NAME exists but has no associated MC resource group"
+        deleteCluster $CLUSTER_NAME $RESOURCE_GROUP_NAME
         create_cluster="true"
+    else
+        # The cluster exists and has an associated MC resource group
+        provisioning_state=$(az aks show -n $CLUSTER_NAME -g $RESOURCE_GROUP_NAME -ojson | jq '.provisioningState' | tr -d "\"")
```

Comment:
nit for refactor: we can probably move this check for provisioning state outside, as soon as we check if the cluster exists or not. 

following that, 
if exists == false or provisioning state == failed
    delete cluster
fi

saves the redundant calls from L35-37 but can follow up later.

Reply:
yep, I think I originally planned on going down this path, but figured we could avoid an unnecessary call to "aks show" - although I think the readability benefits/code reduction do trump that :)